### ISSUE TITLE
Parts of file path ignored when verifying hashes

### ISSUE THREAD

**Issue by Markus74a** [rhash/RHash#139]
OS. Windows 10
RHash: 1.4.0, build from latest source (*8f6b882*)

When using the `-c` verify, some paths containing spaces will get accessed incorrectly:
```
C:\test>dir /s /b /a
C:\test\AABBCC DDEEFFGGHH
C:\test\AABBCCD DEEFFGGHH
C:\test\AABBCCDD EEFFGGHH
C:\test\AABBCCDDEE FFGGHH
C:\test\AABBCC DDEEFFGGHH\ffffgggg hhhhiiii.txt
C:\test\AABBCCD DEEFFGGHH\ffffgggg hhhhiiii.txt
C:\test\AABBCCDD EEFFGGHH\ffffgggg hhhhiiii.txt
C:\test\AABBCCDDEE FFGGHH\ffffgggg hhhhiiii.txt

C:\test>rhash --sha1 -r *.* -o hash.sha1

C:\test>rhash --sha1 -c hash.sha1

--( Verifying hash.sha1 )-------------------------------------------------------
AABBCC DDEEFFGGHH\ffffgggg hhhhiiii.txt             OK
DEEFFGGHH\ffffgggg hhhhiiii.txt                     No such file or directory
EEFFGGHH\ffffgggg hhhhiiii.txt                      No such file or directory
AABBCCDDEE FFGGHH\ffffgggg hhhhiiii.txt             OK
--------------------------------------------------------------------------------
Errors Occurred: Errors:0   Miss:2   Success:2   Total:4

C:\test>type hash.sha1
a9993e364706816aba3e25717850c26c9cd0d89d  AABBCC DDEEFFGGHH\ffffgggg hhhhiiii.txt
a9993e364706816aba3e25717850c26c9cd0d89d  AABBCCD DEEFFGGHH\ffffgggg hhhhiiii.txt
a9993e364706816aba3e25717850c26c9cd0d89d  AABBCCDD EEFFGGHH\ffffgggg hhhhiiii.txt
a9993e364706816aba3e25717850c26c9cd0d89d  AABBCCDDEE FFGGHH\ffffgggg hhhhiiii.txt
```
I'm specifying `--sha1`, so there should be no need to guess if `AABBCCD` or `AABBCCDD` is some kind of other hash.

**Comment by rhash** [comment]
Thanks for reporting this issue. The case is a bit complicated. For historical reasons RHash tries to find several message digests or checksums before the filepath. So AABBCCDD is parsed as a CRC32 checksum and AABBCCD as a BASE32(CRC32) string, same as in #133. 

Since this is intended behaviors, it can't be considered as a bug, even if it is unusual for many users. On the other hand, sha1 and md5 formats are very popular, so this historical behavior should be changed. I believe the best solution should be:

- RHash after parsing the first message digest or checksum, should try if the rest corresponds to a filepath of an existing file, and if not continue the checksum parsing.

**Comment by rhash** [comment]
Actually this is bug, cause the rhash could not verify some file created by itself or by sha1sum. Thank you once again for the bug report! 

The bug is now fixed by 9105d546ea636b994337b71c976e1606306b1989.
